site.py

```diff
@@ -2,54 +2,61 @@
 """
 
 import sys
-if sys.version_info[0] <3 :
-    raise RuntimeError('We love Python 3.')
-
 import os
-
 from flask import Flask, render_template, url_for, Blueprint
-from flask import redirect, abort
+from flask import abort
 from flask_frozen import Freezer
-import yaml
-
 from elsa import cli
 
+if sys.version_info[0] < 3:
+    raise RuntimeError('We love Python 3.')
+
 app = Flask('naucsepythoncz', template_folder="")
 app.config['TEMPLATES_AUTO_RELOAD'] = True
 
-COURSES = os.listdir(app.root_path+"/courses/")
+COURSES = os.listdir(app.root_path + "/courses/")
 
 # Hacky blueprints for static directories
 
 for course in COURSES:
-    blueprint = Blueprint('course_'+course, __name__,
-        static_url_path='/courses/'+course+'/static',
-        static_folder=app.root_path+"/courses/"+course+"/static")
+    blueprint = Blueprint('course_' + course, __name__,
+        static_url_path='/courses/' + course + '/static',
+        static_folder=app.root_path + "/courses/" + course + "/static")
     app.register_blueprint(blueprint)
 
 
 def course_url(path):
     path = path.rstrip('/')
     return url_for('course', path=path)
 
+
 app.jinja_env.globals['course_url'] = course_url
 
+
 @app.route('/')
 def index():
     return render_template("templates/index.html")
 
+
 @app.route('/courses/<path:path>/')
 def course(path):
-    template_paths = ["courses/"+path+'.html', "courses/"+path+'/index.html']
+    template_paths = ["courses/" + path + '.html', "courses/" + path + '/index.html']
     for template_path in template_paths:
         if os.path.exists(os.path.join(app.root_path, template_path)):
             return render_template(template_path)
-    
-    
+
     abort(404)
 
+
+@app.route('/courses/<course_name>/static/<file_name>')
+def course_file(path):
+    file = 'courses/' + course_name + '/static/' + file_name
+    if os.path.exists(os.path.join(app.root_path, file)):
+        return file
+    abort(404)
```

Comment:
Correct, @encukou.  Serving files like this isn't a good idea, for one because Flask will not serve the correct MIME type (but don't quote me on this).  The Blueprints created automatically above should handle static files for course directories already.